server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -680,10 +681,14 @@ public IPAddressVO doInTransaction(TransactionStatus status) throws Insufficient
 
         // If owner has dedicated Public IP ranges, fetch IP from the dedicated range
         // Otherwise fetch IP from the system pool
-        List<AccountVlanMapVO> maps = _accountVlanMapDao.listAccountVlanMapsByAccount(owner.getId());
-        for (AccountVlanMapVO map : maps) {
-            if (vlanDbIds == null || vlanDbIds.contains(map.getVlanDbId()))
-                dedicatedVlanDbIds.add(map.getVlanDbId());
+        Network network = _networksDao.findById(guestNetworkId);
+        //Checking if network is null in the case of system VM's. At the time of allocation of IP address to systemVm, no network is present.
+        if(network == null || !(network.getGuestType() == GuestType.Shared && zone.getNetworkType() == NetworkType.Advanced)) {
```

Comment:
Hi, kansal.
Could you extract this 'if' content to a method with a little test case and a Javadoc? 
You had explained this 'if' pretty good in the comment above, but I think that is a little confuse to understand yet, I know that you can explain this if better and use a Javadoc to do it.

Ty.